yet_another_bioinformatic_tool.py

```diff
@@ -1,142 +1,319 @@
+        )
+
+    def __repr__(self) -> str:
+        return (
+            f"FastQFilter(\n\tpath_to_input='{self.path_to_input}',"
+            + f"\n\tgc_bounds={self.gc_bounds},"
+            + f"\n\tlength_bounds={self.length_bounds},"
+            + f"\n\tquality_threshold={self.quality_threshold}\n)"
+        )
+
+
+class BiologicalSequence(str, ABC):
+    """
+    Abstract base class representing a biological sequence.
+
+    Attributes
+    ----------
+    alphabet : set
+        The set of valid characters in the sequence.
+
+    Methods
+    -------
+    check_alphabet(alphabet: Iterable = None) -> bool
+        Checks if the sequence characters are within the given alphabet.
     """
-    Accepts command and runs it on input data with params
 
-    Arguments:
-    - seqs (str): Input in form of path, seq, seq list or seq dct
-    - command (str): Valid command from command_dct
-    - **kwargs to be passed to inner funcs
+    @property
+    @abstractmethod
+    def alphabet(self) -> set[str]:
+        pass
+
+    def check_alphabet(self, alphabet: Iterable[str] | None = None) -> bool:
+        if alphabet is None:
+            alphabet = self.alphabet
+        return set(self).issubset(set(alphabet))
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Abstract base class representing a nucleic acid sequence.
 
-    Return:
-    - output_dct (dict): 
-        dict where keys are number or name of seq and values are results of command run
+    Methods
+    -------
+    complement() -> Self
+        Returns the complement of the sequence.
+    gc_content() -> float
+        Calculates the GC content of the sequence.
     """
 
-    output_dict = {}
-    for seq_name, seq in seqs.items():
-        if command in command_dct_prot:
-            if command == 'is_protein_valid':
-                output_dict |= {seq_name: protein_tools.is_protein_valid(seq)}
-            else:
-                if protein_tools.is_protein_valid(seq):
-                    output_dict |= {seq_name:
-                                    command_dct_prot[command](seq, **kwargs)}
-                else:
-                    raise ValueError(f'Invalid protein, name/number: {seq_name}')
-
-    if len(output_dict) == 1:
-        return output_dict[list(output_dict.keys())[0]]
-    return output_dict
+    @abstractmethod
+    def _comp_dct(self) -> None:
+        self._check_implementation()
+        pass
+
+    @property
+    def alphabet(self) -> set[str]:
+        self._check_implementation()
+        return set(self._comp_dct)
+
+    def complement(self: Self) -> Self:
+        self._check_implementation()
+        return self.__class__("".join([self._comp_dct[letter] for letter in self]))
+
+    @property
+    def gc_content(self) -> float:
+        self._check_implementation()
+        gc_count = 0
+        for letter in self:
+            if letter.lower() == "g" or letter.lower == "c":
```

Comment:
```suggestion
            if letter.lower() == "g" or letter.lower() == "c":
```
 Самое коварное, что код не выдаст ошибки, потому что `.lower` это `<built-in method lower of str object at ()>` и объект можно сравнить со строкой, это просто всегда False.

Минус 1 балл.